x-pack/legacy/plugins/maps/public/embeddable/map_embeddable.js

```diff
@@ -244,5 +248,15 @@ export class MapEmbeddable extends Embeddable {
         openTOCDetails,
       });
     }
+
+    const hiddenLayerIds = getHiddenLayers(this._store.getState())
+      .map(layer => layer.id)
```

Comment:
Can you move .map and .sort to the selector? The selector should return the data in the format needed here.